File: decompiled/c/5ddbe1.c

void FUN_005ddbe1(void)

{
  byte in_AL;
  byte bVar1;
  byte in_AH;
  char cVar2;
  char *pcVar3;
  uint uVar4;
  
  uVar4 = (uint)in_AL;
  bVar1 = 1;
  pcVar3 = &DAT_0065ea78;
  do {
    cVar2 = (&DAT_005f6b10)[uVar4 * 4];
    if (bVar1 == in_AH) {
      if ((&DAT_005f6b13)[uVar4 * 4] != -1) {
        cVar2 = (&DAT_005f6b13)[uVar4 * 4];
      }
    }
    if (bVar1 == 1) {
      if ((&DAT_005f6b11)[uVar4 * 4] != -1) {
        cVar2 = (&DAT_005f6b11)[uVar4 * 4];
      }
    }
    if (bVar1 == 2) {
      if ((&DAT_005f6b12)[uVar4 * 4] != -1) {
        cVar2 = (&DAT_005f6b12)[uVar4 * 4];
      }
    }
    *pcVar3 = cVar2;
    pcVar3 = pcVar3 + 1;
    bVar1 = bVar1 + 1;
  } while (bVar1 <= in_AH);
  *pcVar3 = -1;
  return;
}

File: ported/auto/5ddbe1.js
// Auto-translated from Ghidra C by tools/c-to-js/translate.js.
// Source: decompiled/c/5ddbe1.c
// Edit by hand only after diff-test passes — re-running the translator will overwrite.

/** @typedef {import("../../runtime/heap.js").Heap} Heap */

import { regs } from "../../runtime/regs.js";
export function FUN_005ddbe1(heap) {
  let in_AL = regs.eax & 0xff;
  let bVar1 = 0;
  let in_AH = (regs.eax >>> 8) & 0xff;
  let cVar2 = 0;
  let pcVar3 = 0;
  let uVar4 = 0;
  uVar4 = ((((in_AL) >>> 0)) >>> 0);
  bVar1 = ((1) & 0xff);
  pcVar3 = ((0x0065ea78) >>> 0);
  do {
    cVar2 = ((heap.u32((0x005f6b10) + (uVar4 * 4) * 4)) & 0xff);
    if (bVar1 == in_AH) {
      if ((heap.u32((0x005f6b13) + (uVar4 * 4) * 4) | 0) != -1) {
        cVar2 = ((heap.u32((0x005f6b13) + (uVar4 * 4) * 4)) & 0xff);
      }
    }
    if (bVar1 == 1) {
      if ((heap.u32((0x005f6b11) + (uVar4 * 4) * 4) | 0) != -1) {
        cVar2 = ((heap.u32((0x005f6b11) + (uVar4 * 4) * 4)) & 0xff);
      }
    }
    if (bVar1 == 2) {
      if ((heap.u32((0x005f6b12) + (uVar4 * 4) * 4) | 0) != -1) {
        cVar2 = ((heap.u32((0x005f6b12) + (uVar4 * 4) * 4)) & 0xff);
      }
    }
    heap.setU32(pcVar3, (cVar2) & 0xffffffff);
    pcVar3 = ((pcVar3 + 1) >>> 0);
    bVar1 = ((bVar1 + 1) & 0xff);
  } while (bVar1 <= in_AH);
  heap.setU32(pcVar3, (-1) & 0xffffffff);
  return;
}
